.github/workflows/publish-gh-pages.yml

```
@@ -0,0 +1,54 @@
+name: Publish Github Pages
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        #python version is required in setup the python environment
+        required: true
+        type: string
+
+jobs:
+  publish-docs:
+    name: "Publish Github Pages"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      # Checkout the code
+      - name: Checkout Source
+        uses: actions/checkout@v2
```

There's a newer version, so I would suggest going with that
```suggestion
        uses: actions/checkout@v3
```